### ISSUE TITLE
Add config for customizing cache headers for use with different hosting providers

### ISSUE THREAD

**Issue by pmi** [enonic/lib-nextjs#148]
It's the Vercel only who is supported currently and it is hardcoded

**Comment by alansemenov** [comment]
Add more details on what the issue is about

**Comment by pmi** [comment]
We currently have a condition: `const isVercelPrerender = response.headers['x-vercel-cache'] === 'PRERENDER';`

Needed to know that `log.debug('Vercel returned static content instead of preview, the token had most likely expired');`

That will not work for other hosting services, but on the other hand they might not have a problem of returning static version of content for requests with expired preview token that vercel has.

**Comment by alansemenov** [comment]
Even if they do have a similar problem, we don't know how they work around it, ie we don't know if they are sending a header like Vercel does and what's the value in that header. I think we should wait with this task until we have a specific vendor that we can test our integration against.

@sigdestad?

**Comment by sigdestad** [comment]
Didn't we figure out that Next.js will return a "Preview" header if you are actually getting the preview? Using this we would not need any special handling at all AFAIK?

**Comment by sigdestad** [comment]
@pmi ?

**Comment by pmi** [comment]
Hmm, I think we discussed that, but there is a problem with headers.
We need to get a hold of response to set them, which we can not do because even when in preview mode we are still using same code that is used for static content.

**Comment by sigdestad** [comment]
Not sure if I follow. Doing a req for Next.js it will return a response, if it is in "preview" mode that will be stated in the header - and you will know if you hit static content or not?

**Comment by sigdestad** [comment]
Then, there should be no need for checking custom headers like Vercel.

**Comment by pmi** [comment]
We can not set a header on a response because we use static generation.
There is no access to response in both static and preview modes.

**Comment by pmi** [comment]
And on top of that, request doesn't even make it to nextjs server on Vercel, but it is rather handled by Vercel's cache/wrapper 😞

**Comment by sigdestad** [comment]
I am not talking about setting a response header, I'm talking about reading the standard response headers from Next in the proxy. If next is in preview mode - it will contain a header with this information. If not, it is static, or token has expired or similar.

**Comment by pmi** [comment]
According to the [issue description](https://github.com/enonic/lib-nextjs/issues/146) Vercel still returns old token along with static page which was the main problem, because nextjs should've returned empty header saying that we need to renew the token.

**Comment by sigdestad** [comment]
Sure, but could you not simply change to check if you get the "preview" header instead? Then attempt to renew if it is not there? This would work regardless of Vercel or not?

**Comment by pmi** [comment]
No, because the problem is that we ALWAYS get preview header from Vercel, even when token has expired and we expect nextjs to return an empty header.

**Comment by sigdestad** [comment]
Ok, not good. That could have been mentioned earlier :-)

Maybe file an issue with Vercel?
